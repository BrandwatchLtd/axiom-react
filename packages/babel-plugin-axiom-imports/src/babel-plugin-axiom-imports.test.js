import path from 'path';
import { transformFileSync } from 'babel-core';
import babelPluginAxiom from './babel-plugin-axiom-imports';
import babelPluginTransformSvgAxiom from '@brandwatch/babel-plugin-transform-svg-axiom';

describe('babelPluginAxiom', () => {
  test('aliased imports', () => {
    expect(transformFileSync(path.resolve(__dirname, '../test/alias.js'), {
      plugins: [babelPluginAxiom],
    }).code).toMatchSnapshot();
  });

  test('all imports', () => {
    expect(transformFileSync(path.resolve(__dirname, '../test/all.js'), {
      plugins: [babelPluginAxiom],
    }).code).toMatchSnapshot();
  });

  test('used as defaults', () => {
    expect(transformFileSync(path.resolve(__dirname, '../test/default.js'), {
      plugins: [babelPluginAxiom],
    }).code).toMatchSnapshot();
  });

  test('direct import', () => {
    expect(transformFileSync(path.resolve(__dirname, '../test/direct.js'), {
      plugins: [babelPluginAxiom],
    }).code).toMatchSnapshot();
  });

  test('destructured imports', () => {
    expect(transformFileSync(path.resolve(__dirname, '../test/destructure.js'), {
      plugins: [babelPluginAxiom],
    }).code).toMatchSnapshot();
  });

  test('scoped imports', () => {
    expect(transformFileSync(path.resolve(__dirname, '../test/scoped.js'), {
      plugins: [babelPluginAxiom],
    }).code).toMatchSnapshot();
  });

  test('property access', () => {
    expect(transformFileSync(path.resolve(__dirname, '../test/property.js'), {
      plugins: [babelPluginTransformSvgAxiom, babelPluginAxiom],
    }).code).toMatchSnapshot();
  });
});
