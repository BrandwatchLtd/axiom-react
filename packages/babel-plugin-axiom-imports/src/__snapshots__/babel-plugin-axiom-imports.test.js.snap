// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babelPluginAxiom aliased imports 1`] = `
"\\"use strict\\";

/* eslint-disable no-console */
var shortDate = function shortDate() {};

var longDate = function longDate() {};

var Base = function Base() {};

var DotPlotChart = function DotPlotChart() {};

var findComponent = function findComponent() {};

console.log(shortDate);
console.log(longDate);
console.log(Base);
console.log(DotPlotChart);
console.log(findComponent);
console.log(shortDateAxiom);
console.log(longDateAxiom);
console.log(BaseAxiom);
console.log(DotPlotChartAxiom);
console.log(findComponentAxiom);"
`;

exports[`babelPluginAxiom all imports 1`] = `
"\\"use strict\\";

function _typeof(obj) { \\"@babel/helpers - typeof\\"; if (typeof Symbol === \\"function\\" && typeof Symbol.iterator === \\"symbol\\") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === \\"function\\" && obj.constructor === Symbol && obj !== Symbol.prototype ? \\"symbol\\" : typeof obj; }; } return _typeof(obj); }

var materials = _interopRequireWildcard(require(\\"@brandwatch/axiom-materials\\"));

function _getRequireWildcardCache() { if (typeof WeakMap !== \\"function\\") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== \\"object\\" && typeof obj !== \\"function\\") { return { \\"default\\": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj[\\"default\\"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/* eslint-disable no-console */
console.log(materials.colors);"
`;

exports[`babelPluginAxiom destructured imports 1`] = `
"\\"use strict\\";

/* eslint-disable no-console */
console.log(shortDate);
console.log(longDate);
console.log(Base);
console.log(DotPlotChart);
console.log(findComponent);
console.log(colors);
console.log(colors.productColorNames);"
`;

exports[`babelPluginAxiom direct import 1`] = `
"\\"use strict\\";

var _axiomMaterials = _interopRequireDefault(require(\\"@brandwatch/axiom-materials\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { \\"default\\": obj }; }

/* eslint-disable no-console */
console.log(_axiomMaterials[\\"default\\"].colors);"
`;

exports[`babelPluginAxiom property access 1`] = `
"\\"use strict\\";

/* eslint-disable no-console */
console.log(colors.foo);
console.log(colors[\\"foo\\"]);
console.log(colors[\\"foo-bar\\"]);"
`;

exports[`babelPluginAxiom scoped imports 1`] = `
"\\"use strict\\";

/* eslint-disable no-console */
var foo = function foo(findComponent) {
  return findComponent();
};

var bar = function bar() {
  return findComponent();
};

foo(findComponent);
bar();"
`;

exports[`babelPluginAxiom used as defaults 1`] = `
"\\"use strict\\";

/* eslint-disable no-console */
var productColors = Object.keys(colors.productColorNames);

var fn = function fn() {
  var colors = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : productColors;
  console.log(colors);
};

fn();"
`;
